modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java

```diff
@@ -357,6 +342,59 @@ countSearchRequest, _getSearchRequest(searchContext), query,
 		return countSearchRequest;
 	}
 
+	private PointInTime _createPointInTime(
+		SearchContext searchContext, SearchRequest searchRequest) {
+
+		OpenPointInTimeRequest openPointInTimeRequest =
+			new OpenPointInTimeRequest(1);
+
+		openPointInTimeRequest.setIndices(
+			_getIndexes(searchRequest, searchContext));
+
+		OpenPointInTimeResponse openPointInTimeResponse =
+			_searchEngineAdapter.execute(openPointInTimeRequest);
+
+		PointInTime pointInTime = new PointInTime(
+			openPointInTimeResponse.pitId());
+
+		// this should be configurable
+
+		pointInTime.setKeepAlive("10m");
+
+		return pointInTime;
+	}
+
+	private SearchSearchRequest _createSearchSearchRequestWithDeepPagination(
+		SearchRequest searchRequest, SearchContext searchContext, Query query) {
+
+		SearchSearchRequest searchSearchRequest = createSearchSearchRequest(
+			searchRequest, searchContext, query);
+
+		searchSearchRequest.setPointInTime( // GL: should be configurable
+			_createPointInTime(searchContext, searchRequest));
+
+		Sort[] kernelSearchSorts = searchContext.getSorts();
+		List<com.liferay.portal.search.sort.Sort> sorts =
+			searchRequest.getSorts();
+
+		if ((kernelSearchSorts == null) && sorts.isEmpty()) {
```

Comment:
Still not sure about this: I commented out lines 380-393 and still got the results with empty sorts. Was there something else I missed?

But in case this code  stayed:
- no need to declare sorts and kernelSearchSorts at all just use ArrayUtil and ListUtil isEmpty() on 380
- on line 392-393 I think the order is wrong. kernel search sorts firts
- on line 392-393 the sorts are set to original values regardless of block 380-390

Reply:
Did you test more than 10.000 documents? without the sorts it not works

Reply:
Ok. The logic doesn't still look correct to me. Maybe just:

```

		if (ArrayUtil.isEmpty(searchContext.getSorts()) ||
				ListUtil.isEmpty(searchRequest.getSorts())) {
			
			ScoreSort scoreSort = _sorts.score();

			scoreSort.setSortOrder(SortOrder.DESC);
			
			searchSearchRequest.addSorts(scoreSort, _sorts.field("_shard_doc"));
		} else {
			searchSearchRequest.setSorts(searchContext.getSorts());
			searchSearchRequest.setSorts(searchRequest.getSorts());
		}
```

Reply:
for some reason https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/java_else_statement_check.markdown#javaelsestatementcheck is removing the else statement

Reply:
Looks correct now. Assuming SF won't allow 'else', I'd still however further condense it a bit by remove creating the sorts variables (lookup is not costly) and just:

	private SearchSearchRequest _createSearchSearchRequestWithDeepPagination(
		SearchRequest searchRequest, SearchContext searchContext, Query query) {

		SearchSearchRequest searchSearchRequest = createSearchSearchRequest(
			searchRequest, searchContext, query);

		searchSearchRequest.setPointInTime( 
			_createPointInTime(searchContext, searchRequest));

		if (ArrayUtil.isEmpty(searchContext.getSorts()) || 
				ListUtil.isEmpty(searchRequest.getSorts())) {

			ScoreSort scoreSort = _sorts.score();

			scoreSort.setSortOrder(SortOrder.DESC);
			
			searchSearchRequest.addSorts(scoreSort, _sorts.field("_shard_doc"));
			
			return searchSearchRequest;
		}
		
		searchSearchRequest.setSorts(searchContext.getSorts());
		searchSearchRequest.setSorts(searchRequest.getSorts());
		
		return searchSearchRequest;
	}